### PR TITLE
Support Python 3.13

### PR DESCRIPTION
**Pull request by andersonhc**
e.g. Fixes #0 

**Checklist**:

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -16,11 +16,11 @@ classifiers =
   Intended Audience :: Developers
   License :: OSI Approved :: GNU Lesser General Public License v3 (LGPLv3)
   Programming Language :: Python
-  Programming Language :: Python :: 3.8
```
Comment (Lucas-C) [inline setup.cfg:19]: Why dropping support for Python 3.8?

**Comment by Lucas-C** [comment]
Currently, this seems blocked by this issue: https://github.com/pylint-dev/pylint/issues/10112

**Review by Lucas-C — APPROVED** [review]

**Comment by Lucas-C** [comment]
I think now this is blocked by the failing test I'm trying to fix in https://github.com/py-pdf/fpdf2/pull/1321 🙂

File: setup.cfg (hunk 1 above)
Comment (andersonhc) [reply setup.cfg]: I reverted it, was trying to check the CI run time difference without 2 extra runners

**Comment by Lucas-C** [comment]
I squashed the 11 commits on this branch and rebased it.

**Review by Lucas-C — APPROVED** [review]

**Comment by Lucas-C** [comment]
Now it seems that `pytest` is crashing with Python 3.12 under Windows 😢
```
  pytest -vv --trace-memory-usage
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    pythonLocation: C:\hostedtoolcache\windows\Python\3.13.1\x64
    PKG_CONFIG_PATH: C:\hostedtoolcache\windows\Python\3.13.1\x64/lib/pkgconfig
    Python_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.13.1\x64
    Python2_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.13.1\x64
    Python3_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.13.1\x64
    CHECK_EXEC_TIME: 
    CHECK_RSS_MEMORY: test-enabled
============================= test session starts =============================
platform win32 -- Python 3.13.1, pytest-8.3.4, pluggy-1.5.0 -- C:\hostedtoolcache\windows\Python\3.13.1\x64\python.exe
cachedir: .pytest_cache
rootdir: D:\a\fpdf2\fpdf2
configfile: setup.cfg
plugins: cov-6.0.0
Windows fatal exception: access violation

Current thread 0x000015d4 (most recent call first):
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\numpy\core\getlimits.py", line 52 in __init__
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\numpy\core\getlimits.py", line 230 in _register_known_types
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\numpy\__init__.py", line 235 in <module>
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1026 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\camelot\core.py", line 10 in <module>
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1026 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\camelot\handlers.py", line 8 in <module>
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1026 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\camelot\io.py", line 5 in <module>
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1026 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\camelot\__init__.py", line 6 in <module>
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1026 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "D:\a\fpdf2\fpdf2\test\table\test_table_extraction.py", line 9 in <module>
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\assertion\rewrite.py", line 184 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "<frozen importlib._bootstrap>", line 1387 in _gcd_import
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\importlib\__init__.py", line 88 in import_module
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\pathlib.py", line 587 in import_path
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\python.py", line 493 in importtestmodule
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\python.py", line 546 in _getobj
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\python.py", line 284 in obj
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\python.py", line 562 in _register_setup_module_fixture
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\python.py", line 549 in collect
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\runner.py", line 389 in collect
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\runner.py", line 341 in from_call
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\runner.py", line 391 in pytest_make_collect_report
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_callers.py", line 103 in _multicall
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_hooks.py", line 513 in __call__
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\runner.py", line 567 in collect_one_node
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 835 in _collect_one_node
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 970 in genitems
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 975 in genitems
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 975 in genitems
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 975 in genitems
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 809 in perform_collect
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 347 in pytest_collection
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_callers.py", line 103 in _multicall
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_hooks.py", line 513 in __call__
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 336 in _main
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 283 in wrap_session
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\main.py", line 330 in pytest_cmdline_main
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_callers.py", line 103 in _multicall
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\pluggy\_hooks.py", line 513 in __call__
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\config\__init__.py", line 175 in main
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Lib\site-packages\_pytest\config\__init__.py", line 201 in console_main
  File "C:\hostedtoolcache\windows\Python\3.13.1\x64\Scripts\pytest.exe\__main__.py", line 7 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main
collecting ... 
```

**Comment by Lucas-C** [comment]
Thank you @andersonhc 👍

**Pull request merged**